pennylane/transforms/decompositions.py

```diff
@@ -0,0 +1,114 @@
+
+    # Check unitarity
+    if not qml.math.allclose(
+        qml.math.dot(U, qml.math.T(qml.math.conj(U))), qml.math.eye(2), atol=1e-7
+    ):
+        raise ValueError("Operator must be unitary.")
+
+    # Compute the determinant
+    det = U[0, 0] * U[1, 1] - U[0, 1] * U[1, 0]
+
+    # Convert to SU(2) if it's not close to 1
+    if not qml.math.allclose(det, [1.0]):
+        exp_angle = -1j * qml.math.cast_like(qml.math.angle(det), 1j) / 2
+        U = qml.math.cast_like(U, exp_angle) * qml.math.exp(exp_angle)
+
+    return U
+
+
+def _zyz_decomposition(U, wire):
+    r"""Helper function to recover the rotation angles of a single-qubit matrix :math:`U`.
+
+    The set of angles are chosen so as to implement :math:`U` up to a global phase.
+
+    Args:
+        U (tensor): A 2 x 2 unitary matrix.
+        tol (float): The tolerance at which an angle is considered to be close enough
+            to 0. Needed to deal with varying precision across interfaces.
+
+    Returns:
+        (float, float, float): A set of angles (:math:`\phi`, :math:`\theta`, :math:`\omega`)
+        that implement U as a sequence :math:`U = RZ(\omega) RY(\theta) RZ(\phi)`.
+    """
+    U = _convert_to_su2(U)
+
+    # Check if the matrix is diagonal; only need to check one corner.
+    # If it is diagonal, we don't need a full Rot, just return an RZ.
+    if qml.math.allclose(U[0, 1], [0.0]):
+        omega = 2 * qml.math.angle(U[1, 1])
+        return [qml.RZ(omega, wires=wire)]
+
+    # If not diagonal, compute the angle of the RY
+    cos2_theta_over_2 = qml.math.abs(U[0, 0] * U[1, 1])
+    theta = 2 * qml.math.arccos(qml.math.sqrt(cos2_theta_over_2))
+
+    # If the top left element is 0, can only use the off-diagonal elements We
+    # have to be very careful with the math here to ensure things that get
+    # multiplied together are of the correct type in the different interfaces.
+    if qml.math.allclose(U[0, 0], [0.0]):
+        phi = 1j * qml.math.log(U[0, 1] / U[1, 0])
+        omega = -phi - qml.math.cast_like(2 * qml.math.angle(U[1, 0]), phi)
+    else:
+        el_division = U[0, 0] / U[1, 0]
+        tan_part = qml.math.cast_like(qml.math.tan(theta / 2), el_division)
+        omega = 1j * qml.math.log(tan_part * el_division)
+        phi = -omega - qml.math.cast_like(2 * qml.math.angle(U[0, 0]), omega)
+
+    return [qml.Rot(qml.math.real(phi), qml.math.real(theta), qml.math.real(omega), wires=wire)]
+
+
+@qfunc_transform
+def decompose_single_qubit_unitaries(tape):
+    """Quantum function transform to decomposes all instances of single-qubit QubitUnitary
+    operations to a sequence of rotations of the form ``RZ``, ``RY``, ``RZ``.
+
+    Args:
+        tape (qml.tape.QuantumTape): A quantum tape.
+    """
+    for op in tape.operations + tape.measurements:
+        if isinstance(op, qml.QubitUnitary):
+            dim_U = qml.math.shape(op.parameters[0])[0]
+
+            if dim_U != 2:
+                continue
```

Comment:
Actually I think the checks here are the ones to disable since this checked in `_zyz_decomposition` via `_convert_to_su2`. I think the check needs to stay in there because that's what gets called by the `QubitUnitary.decomposition` method (which itself doesn't do a full check for unitarity).